javascriptlib/src/mill/javascriptlib/TypeScriptModule.scala

```diff
@@ -110,44 +174,106 @@ trait TypeScriptModule extends Module { outer =>
       ).mkString(":"))
     }
 
-  // define computed arguments and where they should be placed (before/after user arguments)
-  def computedArgs: Task[Seq[String]] = Task { Seq.empty[String] }
+  def computedArgs: T[Seq[String]] = Task { Seq.empty[String] }
+
+  def executionFlags: T[Map[String, String]] = Task { Map.empty[String, String] }
 
   def run(args: mill.define.Args): Command[CommandResult] = Task.Command {
     val mainFile = mainFilePath()
     val tsnode = npmInstall().path / "node_modules/.bin/ts-node"
     val tsconfigpaths = npmInstall().path / "node_modules/tsconfig-paths/register"
-    val env = mkENV() + ("RESOURCES" -> resources().path.toString)
+    val env = forkEnv()
+
+    val execFlags: Seq[String] = executionFlags().map {
+      case (key, "") => s"--$key"
+      case (key, value) => s"--$key=$value"
+      case _ => ""
+    }.toSeq
 
     os.call(
-      (tsnode, "-r", tsconfigpaths, mainFile, computedArgs(), args.value),
+      (
+        "node",
+        execFlags,
+        tsnode,
+        "-r",
+        tsconfigpaths,
+        mainFile,
+        computedArgs(),
+        args.value
+      ),
       stdout = os.Inherit,
       env = env,
       cwd = compile()._1.path
     )
   }
 
-  def bundleFlags: T[Map[String, Seq[String]]] = Task { Map.empty[String, Seq[String]] }
+  def bundleExternal: T[Seq[String]] = Task { Seq("'fs'", "'path'") }
+
+  def bundleFlags: T[Map[String, String]] = Task {
+    Map(
+      "platform" -> "'node'",
+      "entryPoints" -> s"['${mainFilePath()}']",
+      "bundle" -> "true"
+    )
+  }
 
   // configure esbuild with @esbuild-plugins/tsconfig-paths
   def bundleScriptBuilder: Task[String] = Task.Anon {
-    val mainFile = mainFilePath()
     val bundle = Task.dest / "bundle.js"
+    val rps = resources().map { p => p.path }.filter(os.exists)
+
+    def envName(input: String): String = {
+      val cleaned = input.replaceAll("[^a-zA-Z0-9]", "") // remove special characters
+      cleaned.toUpperCase
+    }
 
-    val flags = bundleFlags().map { case (key, values) =>
-      s"""  $key: [${values.map(v => s"'$v'").mkString(", ")}],"""
+    val flags = bundleFlags().map { case (key, value) =>
+      s"""  $key: $value,"""
     }.mkString("\n")
 
+    val copyPluginCode =
+      s"""
+         |  plugins: [
+         |  Copy({
+         |    resolveFrom: 'cwd',
+         |    assets: [
+         |      ${rps
+          .map { rp =>
+            val filesToCopy =
+              os.list(rp).filter(os.isFile)
+            filesToCopy
+              .map { file =>
+                s"""{ from: join('$rp', '${file.last}'), to: join('${Task.dest}', '/${rp.last}', '${file.last}') }"""
```

Comment:
We should escape all the interpolated strings here by wrapping them in `ujson.Str(rp)` etc., to ensure things like closing quotes and escaping and newlines are properly handled